### DISCUSSION TITLE
How to remove the dock icon on Mac Tauri?

### DISCUSSION THREAD

**Discussion by billywatkins** [tauri-apps/tauri#8506] [Q&A] [answered]
I am trying to remove the dock icon, close the window, and let the app run in the background using Tauri.

I am able to close the window, but I don't know how to remove the app icon after the window is closed.

```rust
// Prevents additional console window on Windows in release, DO NOT REMOVE!!
#![cfg_attr(not(debug_assertions), windows_subsystem = "windows")]

#[tauri::command]
fn greet(name: &str) -> String {
    format!("Hello ->, {}!", name)
}

fn main() {
    tauri::Builder::default()
        .setup(|app| {
            // #[cfg(target_os = "macos")]
            // app.set_activation_policy(tauri::ActivationPolicy::Accessory); // this removes the dock icon by default
            Ok(())
        })
        .build(tauri::generate_context!())
        .expect("error while building tauri application")
        .run(|_app_handle, event| match event {
// dock icon should be removed here, but I don't know how, share the app state?
            tauri::RunEvent::ExitRequested { api, .. } => {
                api.prevent_exit();
            }
            _ => {}
        })
}

```

**Comment by FabianLars** [comment, marked as answer]
Copy pasting the answer from [stackoverflow](https://stackoverflow.com/a/77737378/13375311) for visibility:

> Tauri doesn't seem to support setting the activation policy after startup, but you can set it yourself at runtime (see the comments and edits to [this answer](https://stackoverflow.com/questions/8317101/add-remove-application-icon-from-dock-at-runtime)) by calling the AppKit API using the `cocoa` crate:

```rust
use cocoa::appkit::{NSApp, NSApplication, NSApplicationActivationPolicy::*};

unsafe {
    let app = NSApp();
    app.setActivationPolicy_(NSApplicationActivationPolicyAccessory);
}
```

> Inserting this immediately after the call to `api.prevent_exit()` causes the window to close and the dock icon to disappear, but the program continues running.

> It's worth noting that Tauri doesn't expect you to do this, so it may cause other unexpected behavior -- it might be difficult to restore the window state later on, for example.

**Reply by billywatkins** [reply]
It works, but when the app icon is leaving the dock, there's a wired terminal icon showing up, replacing the default Tauri icon. Is this expected?

Sorry for this noob question; I am completely new to macOS development.

**Reply by FabianLars** [reply]
I'm not sure to be honest. Does this also happen if you run the app after `tauri build`? I could imagine that this is dev mode specific with macos falling back to the terminal parent process 🤔

**Reply by billywatkins** [reply]
I just found out that it's only happening in dev mode and not after building the app, so it's not a problem

**Reply by billywatkins** [reply]
Hey, sorry to bring this up again, but when I use this approach, the app is not opening from the spotlight search. @FabianLars

**Reply by billywatkins** [reply]
Just found this open PR https://github.com/tauri-apps/tao/pull/517
